services/subscriber/service.go

```diff
@@ -183,20 +183,28 @@ func (s *Service) createSubscription(se subEntry, mode string, destinations []st
 	default:
 		return nil, fmt.Errorf("unknown balance mode %q", mode)
 	}
-	writers := make([]PointsWriter, len(destinations))
-	stats := make([]writerStats, len(writers))
-	for i, dest := range destinations {
+	writers := []PointsWriter{}
+	stats := []writerStats{}
+	// add only valid destinations
+	for _, dest := range destinations {
 		u, err := url.Parse(dest)
 		if err != nil {
-			return nil, err
+			s.Logger.Println("failed to parse subscription url:", dest)
```

Comment:
Log messages should be correct sentences, and `Printf` is more common in the project, i.e.,

```go
s.Logger.Printf("Failed to parse subscription url: %s", dest)
```

Same issue applies to all other log messages in the PR.